[[ Bug 18357 ]] Dispatch command array as argument

bhall2001 opened the pull request:
- Added arrayName to dispatch command syntax
- Added example showing arrayName usage
- Updated description text to mention use of array as an argument

Other fixes not related to bug 18357:
- Added “,” after argumentList in command syntax
- update example usage to show multiple arguments
- Fixed “mesage” to “message” typo in description

bhall2001 committed:
[[ Bug 18357 ]] Dispatch command array as argument

- Added arrayName to dispatch command syntax
- Added example showing arrayName usage
- Updated description text to mention use of array as an argument

Other fixes not related to bug 18357:
- Added “,” after argumentList in command syntax
- update example usage to show multiple arguments
- Fixed “mesage” to “message” typo in description

livecode-vulcan commented:
📝 Hi @bhall2001, I haven't been able to verify that you've signed our [LiveCode Contributor's Agreement](https://livecode.com/account/developer/contribute).

If you have previously signed the Contributor's Agreement, I may not be able to detect it because you haven't [linked your GitHub account to your LiveCode account](https://livecode.com/account/developer/github).

Please see the [information for contributors](https://github.com/runrev/livecode/blob/develop/CONTRIBUTING.md) for more information.

bhall2001 committed:
Add release note

livecodeali commented:
@bhall2001 I don't think some of these changes are necessary, as an array name _is_ an expression which forms part of an argument list. As such the original syntax was correct (and indeed you can pass multiple arrays as parameters to `dispatch` by comma-delimiting them).

I think the best way to fix the bug would be to add a detailed example in which a simple array is created and used as an argument to the dispatch command, and where the target command actually does something with the array parameter it is passed. 

Adding an example which uses two parameters is also a good idea - it would be nice if the command that was `dispatch`ed also existed in the example though.

bhall2001 committed:
Reverted some changes,  expanded examples

- Reverted syntax back to original
- Added sentence stating arrays are valid arguments in argumentList
- Expanded examples

bhall2001 committed:
Expanded examples

- Expanded dispatch command examples

(not sure why this did not go with last commit)

bhall2001 commented:
Not sure what I did wrong with my last commit (sorry, still getting my feet wet with Git) so there's an extra commit. 

The original bug report states there was confusion about the use of arrays in the argumentList. I've added in a sentence in the argumentList definition noting that arrays are valid as arguments.

livecodeali commented:
@livecode-vulcan review ok 755fcfb

livecode-vulcan commented:
💙 review by @livecodeali ok 755fcfb76528610a72b2b648b3eb41f123bfb483

livecode-vulcan commented:
😎 test success 755fcfb76528610a72b2b648b3eb41f123bfb483
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/328)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/333)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/328)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/328)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/334)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/328)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/328)

peter-b commented:
@bhall2001 Thank you for the patch!
